### ISSUE TITLE
Federated share mount point disappears when unavailable

### ISSUE THREAD

**Issue by PVince81** [owncloud/core#17615]
### Steps to reproduce

1. Create two users "user1" and "user2"
1. Create a folder "test"
1. Share "test" with "user2@localhost/owncloud" (or whichever path your local instance has)
1. Login as "user2" and accept the share
1. In the database, change the "remote" value in "oc_share_external" to one of a non-existing server
1. Refresh the page
1. Enter the folder "test"

### Expected result
Folder "test" is still visible.
Entering the folder "test" should show an error message that it is currently not available.

### Actual result
Folder "test" is not present in the list.
This means that a connected sync client would delete all the files, and later redownload them once the share becomes available again.

### Versions
Observed on OC 8.2 git master (1006ec56ceb2c7c5bc14a5401d46767cb4b72e6c)

I found that the "propfind" operation in the Dav class did not convert the exception to StorageNotAvailableException. Furthermore, the code path seemed to come from the scanner which now ignores generic exceptions and removes entries from the result list (could be due to recent changes regarding locking)

CC @icewind1991

I'll bisect this

**Comment by PVince81** [comment]
Also happens on v8.1.0 and on stable8.

Hmmm, then probably not a regression.
I think this happens only because the folder hasn't been scanned yet at the time of the unavailability.

Let me try again with non-scanned and scanned-folder.

**Comment by PVince81** [comment]
Hmmm I don't understand, now I see it happening on 8.0.4 too.

It is supposed to keep the entry. If the entry is requested directly, it should return 503.
But now it doesn't even show it.
I remember testing this many times on older versions and seeing that it worked.
Something must be different in the way I tested it.

**Comment by PVince81** [comment]
It happens here: https://github.com/owncloud/core/blob/v8.1.0/lib/private/files/view.php#L1306

It looks like it only triggers itself when the root cache entry is missing.
So we're still in some kind of pre-scan situation.

**Comment by PVince81** [comment]
Ok, now I understand.

In my test step I changed the URL of the remote share. This means it maps to a different share id, so it's a different cache, which obviously isn't scanned.

I think this is a non-issue, because once the share was scanned once, it will always appear and go through the correct logic.

Closing. Not a regression.

**Comment by SergioBertolinSG** [comment]
Tested using latest stable8.1, working correctly, there is a message warning you that the storage is not available.

**Comment by SergioBertolinSG** [comment]
But, if I refresh the page test folder dissapears, is this expected?

**Comment by PVince81** [comment]
@SergioBertolinSG don't test this, this is not a use case because you're fiddling with the database. It's invalid.

**Comment by PVince81** [comment]
Removing labels, sorry for the confusion.
